astrokat/simulate.py

```diff
@@ -160,6 +160,13 @@ def __init__(self, kat, **kwargs):
         self.time = self.start_time
         self.katpt_current = None
         self.capture_initialised = False
+        # add USE proxy indicators
+        if "instrument" in self.obs_params:
+            instrument_dict = self.obs_params['instrument']
+            if "pool_resources" in instrument_dict:
+                pool_resources = instrument_dict['pool_resources']
+                if 'fbfuse' not in pool_resources:
+                    self.fbf = False
```

Comment:
Do we ever set `self.fbf` to something else?  I am thinking about the case where we are simulating with fbf.  Does `if session.fbf` evaluate to true somehow?  If it is true, what code is run when we call
`if not session.fbf.check_provisioning_beams_complete()`?

Similarly, if instrument or pool resources isn't specified, will `if session.fbf` evaluate to true or false?  I would hope false, but I'm not sure how that would happen.

Reply:
Technically self.fbf should only ever be the returned instance or None. There should not be anything that sets it to false. The fbfuse data proxy wrapper looks for a fbfuse attribute on the kat object. I don't know if the instrument_dict in obs_params emulates that perfectly/reliably.

Reply:
What I can see from our test runs.
If FBFUSE proxy is in the array, an instance is returned, else None

For the offline simulations I am not going to build fake proxies for USEs, according to me this type of interaction with MKAT observations should not exist.
That said, there seem to be a special case here and I needed a way to check if the FBF proxy was included.

0) When FBF is desired to be in the array, it has to added to the `pool_resources` to ensure the observation does not run without it. So I used this as indicator to mimic having an FBFUSE proxy
1) Next I needed a way to add this into the `simulation` code for when you do not have the `katcorelib` functionality.
Here I simply set it to `False` in the offline simulation since it reacts the same, but is the least amount of fiddle with the code.

This should not affect anything, because `simulate.py` code get's ignored on our systems when `mkat_session` is available.